render-ws-spark-client/src/main/java/org/janelia/render/client/spark/pipeline/AlignmentPipelineStepId.java

```diff
@@ -0,0 +1,46 @@
+package org.janelia.render.client.spark.pipeline;
+
+import org.janelia.render.client.spark.MipmapClient;
+import org.janelia.render.client.spark.match.ClusterCountClient;
+import org.janelia.render.client.spark.match.CopyMatchClient;
+import org.janelia.render.client.spark.match.MultiStagePointMatchClient;
+import org.janelia.render.client.spark.multisem.MFOVMontageMatchPatchClient;
+import org.janelia.render.client.spark.multisem.UnconnectedCrossMFOVClient;
+import org.janelia.render.client.spark.newsolver.DistributedAffineBlockSolverClient;
+
+/**
+ * Identifier for a step in a spark alignment pipeline with a convenience {@link #toStepClient()} builder.
+ *
+ * @author Eric Trautman
+ */
+public enum AlignmentPipelineStepId {
+
+    GENERATE_MIPMAPS,
+    DERIVE_TILE_MATCHES,
+    PATCH_MFOV_MONTAGE_MATCHES,
+    FIND_UNCONNECTED_MFOVS,
+    FIND_UNCONNECTED_TILES_AND_EDGES,
+    FILTER_MATCHES,
+    ALIGN_TILES;
+
+    public AlignmentPipelineStep toStepClient()
+            throws UnsupportedOperationException {
```

Comment:
I usually like to give the enum a private field of type `Supplier<T>` instead of a huge switch statement. This requires a bit more effort up front, but makes the `toStepClient` method generic, such that adding a new enum doesn't have to be done in two places. E.g., define a new enum as `GENERATE_MIPMAPS(MipmapClient::new)` and have the `toStepClient` method only do
```Java
this.supplier.get()
```
Then, you can also get rid of the `UnsupportedOperationException`, since this case cannot happen anymore.

Reply:
Great idea - done.